include/alpaka/vec/Vec.hpp

```diff
@@ -404,14 +445,52 @@ namespace alpaka
         ALPAKA_FN_HOST_ACC friend constexpr auto operator>=(Vec const& p, Vec const& q) -> Vec<TDim, bool>
         {
             Vec<TDim, bool> r;
+#if BOOST_COMP_NVCC && BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(11, 3, 0)
+            if(TDim::value > 0)
+#else
             if constexpr(TDim::value > 0)
+#endif
             {
                 for(typename TDim::value_type i = 0; i < TDim::value; ++i)
                     r[i] = p[i] >= q[i];
             }
             return r;
         }
 
+        //! \return The element-wise logical and relation of two vectors.
+        ALPAKA_NO_HOST_ACC_WARNING
+        ALPAKA_FN_HOST_ACC friend constexpr auto operator&&(Vec const& p, Vec const& q) -> Vec<TDim, bool>
+        {
+            Vec<TDim, bool> r;
+#if BOOST_COMP_NVCC && BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(11, 3, 0)
+            if(TDim::value > 0)
+#else
+            if constexpr(TDim::value > 0)
+#endif
+            {
+                for(typename TDim::value_type i = 0; i < TDim::value; ++i)
+                    r[i] = p[i] && q[i];
+            }
+            return r;
+        }
```

Comment:
I was thinking about two things here:

* Either this functionality is too specialized, because it only works for element types convertible to `bool` and even then it might not always make sense, like for `Vec<..., float>` etc. So it should maybe be a function outside the class.
* Or the feature is too restricted, because it assumes the result of `&&` between two element types is `bool`, and we should use something like `Vec<TDim, decltype(std::declval<TVal>() && std::declval<TVal>())>`.

I concluded however, that we can just keep it as is and refactor it when the time comes.

Reply:
> * it only works for element types convertible to `bool` and even then it might not always make sense, like for `Vec<..., float>` etc

I think it works for types `T` for which, given `T t1, t2`, `t1 && t2` is defined and the result is convertible to `bool`.

I wouldn't mind restricting it to those types (with a `static_assert` or some SFINAE magic, since we do not have concepts in c++17) - but then we should do it for all the other operations, in case `*` does not work as expected, etc.

So... yeah, I agree with you, let's keep it like this and improve it or refactor it later, if needed.